_stbt/control_gpl.py

```diff
@@ -1,16 +1,68 @@
+from .logging import debug
+
+
 class HdmiCecControl(object):
+    cecconfig = {}
+    lib = {}
+
+    # Map our recommended keynames (from linux input-event-codes.h) to the
+    # equivalent CEC commands.
+    _KEYNAMES = {
+        "KEY_OK": "14:44:00",
+        "KEY_UP": "14:44:01",
+        "KEY_DOWN": "14:44:02",
+        "KEY_LEFT": "14:44:03",
+        "KEY_RIGHT": "14:44:04",
+        "KEY_BACK": "14:44:0D"
+    }
+
     def __init__(self, device):
         import cec
-        raise NotImplementedError()
+        self.cecconfig = cec.libcec_configuration()
+        self.cecconfig.strDeviceName = "STB-Tester"
+        self.cecconfig.bActivateSource = 0
+        self.cecconfig.deviceTypes.Add(cec.CEC_DEVICE_TYPE_RECORDING_DEVICE)
+        self.cecconfig.clientVersion = cec.LIBCEC_VERSION_CURRENT
+        self.lib = cec.ICECAdapter.Create(self.cecconfig)
+        # print libCEC version and compilation information
+        debug("libCEC version " + self.lib.VersionToString(self.cecconfig.serverVersion) +
+              " loaded: " + self.lib.GetLibInfo())
+
+        if device == 'auto':
```

Comment:
I would suggest making this conditional `device is None`.  We can change the regex below to pass `device=None` if you specify `hdmi-cec:` as your control.  This will be more consistent with the other remotes.